openmc/data/photon.py

```diff
@@ -735,7 +735,8 @@ def export_to_hdf5(self, path, mode='a', libver='earliest'):
                         shell_values.insert(0, None)
                         df = relax.transitions[shell].replace(
                             shell_values, range(len(shell_values)))
-                        sub_group.create_dataset('transitions', data=df.values)
+                        sub_group.create_dataset(
+                            'transitions', data=df.values.astype(float))
```

Comment:
Interesting, didn't notice that before. The problem is that `to_numpy` was just introduced in 0.24.0, so using it would break this for anyone using a previous version.

Reply:
True. Something to keep in mind for the future I suppose.